packages/server/routes/api/guilds.js

```diff
@@ -239,7 +243,126 @@ router.get(
         message: error.message,
       });
     }
-  }
+  },
+);
+
+// Add guild member
+router.post(
+  "/:guildId/members/:userId",
+  AuthController.authenticate,
+  guildIdValidator,
+  userIdValidator,
+  addGuildMemberValidator,
+  async (request, response) => {
+    const result = validationResult(request);
+
+    if (!result.isEmpty()) {
+      return response.status(400).json({
+        status: "fail",
+        data: result.array(),
+      });
+    }
+
+    const data = matchedData(request);
+    const guildId = data.guildId;
+    const userId = data.userId;
+
+    try {
+      return response.status(200).json({
+        status: "success",
+        data: {
+          createdMember: await GuildController.addGuildMember(guildId, userId),
+        },
+      });
+    } catch (error) {
+      return response.status(500).json({
+        status: "error",
+        message: error.message,
+      });
+    }
+  },
+);
+
+// Update guild member's role
+router.put(
+  "/:guildId/members/:userId/roles",
```

Comment:
Guild members can only have one role, and this route only deals with updating a singular guild member's role here, so it'd be better to keep the route singular instead of plural as well.
```
/:guildId/members/:userId/role
```